pkgs/java_http/.gitattributes

```
@@ -0,0 +1,2 @@
+# Hides jnigen generated code in GitHub diffs.
+lib/src/third_party/** linguist-generated
```

Hmm, Is it possible to more narrowly target this?

I agree that marking these existing generated files as generated is a good idea - I understanding the reasoning for using this target. When I saw the path I was expecting to see `linguist-vendored`, not `linguist-generated`.

I think the sole next directory is probably specific enough to assume it's generated. How about

```suggestion
lib/src/third_party/** linguist-vendored
lib/src/third_party/java/** linguist-generated
```

The first line will catch if we happen to add other third party source for some other reason.

---

Ah I see what you mean, just because it's in the `third-party` directory it doesn't necessarily mean it's been generated by `jnigen`. In the future, we could add third-party code to `third_party` that hasn't been generated by `jnigen`.  Therefore, we shouldn't mark the entire `third_party` directory as `linguist-generated` but instead `linguist-vendored`.

The suggestions look good and have been committed :heavy_check_mark:

Thanks @natebosch!

---

> I think the sole next directory is probably specific enough to assume it's generated.

Would this catch the case where we are generating bindings for classes whose package name doesn't begin with `java`. For example, I think `okhttp3.Request` bindings would currently go in `third_party/okhttp3/Request.dart`.

To solve this we could put all the `jnigen` generated bindings in their own directory such as `third_party/generated`. Then `.gitattributes` would be something like:

```gitattributes
lib/src/third_party/** linguist-vendored
lib/src/third_party/generated/** linguist-generated
```

Although, at the moment we are only generating bindings for classes whose package name does begin with `java`. So we might not need to handle this case for now.

> Hmm, Is it possible to more narrowly target this?

Alternatively, we could also consider generating the `jnigen` bindings as a single file such as `native_java_bindings.dart`. Then `.gitattributes` would be something like:

```gitattributes
lib/src/third_party/** linguist-vendored
lib/src/third_party/native_java_bindings.dart linguist-generated
```

@natebosch What are your thoughts? :brain:

---

> To solve this we could put all the `jnigen` generated bindings in their own directory such as `third_party/generated`.

This sounds like the optimal layout, but I'm also on board with _not_ making this change in favor of reduced churn.

I'd be fine with either (in slight preference order):

1. Keeping this PR as is - we can deal with handling other `linguist-generated` content if and when it starts showing up in PRs, which may never happen.
2. Refactor into a `lib/src/third_party/generated/**` so that we never have to worry about adjusting the config for new generated content, which may be unnecessary churn if we never generate for other classes.

---

> Keeping this PR as is - we can deal with handling other linguist-generated content if and when it starts showing up in PRs, which may never happen.

I think we should go ahead with keeping the PR as is, it feels like the best option to me!